src/firefly/js/visualize/ui/FileUploadViewPanel.jsx

```diff
@@ -217,111 +208,62 @@ function getLoadButtonText() {
 }
 
 
-
-
-
 export function resultFail() {
     showInfoPopup('One or more fields are invalid', 'Validation Error');
 }
 
-const getPartCnt= () => currentReport?.parts?.length ?? 1;
-const getFirstPartType= () => currentSummaryModel?.tableData.data[0]?.[1];
-const getFileFormat= () => currentReport?.fileFormat;
-const isRegion= () => getFirstPartType()===FileAnalysisType.REGION;
 
-function isSinglePartFileSupported() {
+function isSinglePartFileSupported(currentSummaryModel) {
+    const isTablesOnly= () => getAppOptions()?.uploadPanelLimit==='tablesOnly';
     const supportedTypes= isTablesOnly() ? TABLES_ONLY_SUPPORTED_TYPES : SUPPORTED_TYPES;
-    return getFirstPartType() && (supportedTypes.includes(getFirstPartType()));
-}
-
-function isFileSupported() {
-    return getFirstPartType() && (SUPPORTED_TYPES.includes(getFirstPartType()) || getFileFormat()===Format.FITS);
+    return getFirstPartType(currentSummaryModel) && (supportedTypes.includes(getFirstPartType(currentSummaryModel)));
 }
 
 
 function getFirstExtWithData(parts) {
+    const isTablesOnly= () => getAppOptions()?.uploadPanelLimit==='tablesOnly';
     return isTablesOnly() ?
         parts.findIndex((p) => p.type.includes(FileAnalysisType.Table)) :
         parts.findIndex((p) => !p.type.includes(FileAnalysisType.HeaderOnly));
 }
 
+/*-----------------------------------------------------------------------------------------*/
 
+const getLoadingFieldName= (groupKey) => getFieldVal(groupKey, uploadOptions) || FILE_ID;
 
-function tablesOnlyResultSuccess(request) {
-    const tableIndices = getSelectedRows(FileAnalysisType.Table);
-    const imageIndices = getSelectedRows(FileAnalysisType.Image);
-
-    if (tableIndices.length>0) {
-        imageIndices.length>0 && showInfoPopup('Only loading the tables, ignoring the images.');
-        sendTableRequest(tableIndices, getFileCacheKey(), Boolean(request.tablesAsSpectrum==='spectrum'));
-        return true;
-    }
-    else {
-        showInfoPopup('You may only upload tables.');
-        return false;
-    }
-}
+//function getNextState(oldState, groupKey) {
+function getNextState(groupKey, oldState) {
+    // because this value is stored in different fields.. so we have to check on what options were selected to determine the active value
 
+    const SUMMARY_TBL_ID = groupKey; //FileUploadAnalysis
+    const DETAILS_TBL_ID = groupKey + '-Details';
+    const UNKNOWN_FORMAT = 'UNKNOWN';
 
-export function resultSuccess(request) {
-    if (isTablesOnly()) return tablesOnlyResultSuccess(request);
-    const fileCacheKey = getFileCacheKey();
+    let currentReport, currentSummaryModel, currentDetailsModel;
 
-    const tableIndices = getSelectedRows(FileAnalysisType.Table);
-    const imageIndices = getSelectedRows(FileAnalysisType.Image);
+    const fieldState = getField(groupKey, getLoadingFieldName(groupKey)) || {};
+    let {analysisResult, message, summaryModel, prevAnalysisResult, report} = fieldState;
```

Comment:
fieldState only had `analysisResult` and `message` in it, but remove and allowed them to be passed as i showed above.